### PR TITLE
Implement math operators

### LINKED ISSUE #2: Math Ops
## Intro:

Minecraft has several custom math types, such as `Vec3d`, `Vec3i`, `Vec2i`, Matrix types, ect. These types often have operations that can be applied on them, such as `Vec3d.add(Vec3d)`. These can be improved in Kotlin through the use of [operator overloading](https://kotlinlang.org/docs/operator-overloading.html) so that operations such as `Vec3d + Ved3d` may be preformed instead.

## Existing work:
- At this time I am unable to locate existing work on such an API

### PR DESCRIPTION
**Pull request by SilverAndro**
Closes #2 

This implements common math operators for Minecraft's vector types as those are heavily used and have well defined operations, specifically:
- Vec2f
- Vec3d
- Vec3f
- Vec3i
- Vector2f
- Vector3d
- Vector4f

The majority of this code has been generated by a script, with the exception of:
- Vector4fOps.kt
- `dot` and `cross` method implementations

### CODE REVIEW

**Review by Oliver-makes-code — APPROVED** [review]
LGTM

File: wrapper/main/src/main/kotlin/org/quiltmc/qkl/wrapper/main/math/Vec2fOps.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+public operator fun Vec2f.times(other: Vector2f): Vec2f {
+    return Vec2f(
+        this.x * other.x,
+        this.y * other.y
+    )
+}
+//endregion
+
+//region Vector specific operators
+public infix fun Vec2f.dot(other: Vec2f): Float {
```
Comment (leocth) [inline wrapper/main/src/main/kotlin/org/quiltmc/qkl/wrapper/main/math/Vec2fOps.kt:116]: What about cross products and other operations (e.g. normalize, magnitude, etc.)?

**Review by leocth — APPROVED** [review]
Looks fine, but don't we also have matrices in the game?

**Comment by SilverAndro** [comment]
Iirc matrices dont have any sort of built in add/subtract/multiply methods? May have skipped over them by accident tbh 😅 

Ill check tomorrow, although ill likely make that its own PR if so

File: wrapper/main/src/main/kotlin/org/quiltmc/qkl/wrapper/main/math/Vec2fOps.kt (hunk 1 above)
Comment (SilverAndro) [reply wrapper/main/src/main/kotlin/org/quiltmc/qkl/wrapper/main/math/Vec2fOps.kt:116]: Two dimensional cross product ends up being a scalar instead so I ended up skipping it at the time, if it makes sense I can add it back though

Normalize could probably be implemented, but magnitude is just a property so I don't see much reason to add it

**Review by NoComment1105 — APPROVED** [review]

**Comment by SilverAndro** [comment]
Ive added `normalize`, `length`, and `lengthSquared` to all vectors that did not already have it defined.

Vector4f had to be `normalized` because it has an existing, mutating function by that name

**Review by Leo40Git — APPROVED** [review]

**Review by PotatoPresident — APPROVED** [review]
Seems good, although it seems to be a lot of non math changes..

**Comment by SilverAndro** [comment]
> Seems good, although it seems to be a lot of non math changes..

Indeed I have no idea where those came from, it seems to have been some issue with how I tried to rebase my changes?

**Comment by SilverAndro** [comment]
I cant see to remove the extra commits from this PR, if anything in those is incorrect I may just have to close this and open a fixed PR

**Comment by SilverAndro** [comment]
Ive noticed a few issues with this that need fixing (its actually broken in a few cases atm) so I'm going to use this opportunity to open a new PR that fixes the git issues in a few days since the FCP likely needs reset anyways, given I need to rework large sections of this (probably about 1/3 of the PR)

**Pull request closed without merging**